src/core/manager.c

```diff
@@ -3740,8 +3742,45 @@ static int build_generator_environment(Manager *m, char ***ret) {
         return 0;
 }
 
+static int manager_execute_generators(Manager *m, char **paths) {
+        _cleanup_strv_free_ char **ge = NULL;
+        const char *argv[] = {
+                NULL, /* Leave this empty, execute_directory() will fill something in */
+                m->lookup_paths.generator,
+                m->lookup_paths.generator_early,
+                m->lookup_paths.generator_late,
+                NULL,
+        };
+        int r;
+
+        r = build_generator_environment(m, &ge);
+        if (r < 0)
+                return log_error_errno(r, "Failed to build generator environment: %m");
+
+        /* We leave /run read-write because that's where the output goes to,
+         * and we leave /sys as-is, because our code checks whether it is read-only
+         * to detect containerized execution environments. */
+        r = bind_remount_recursive("/", MS_RDONLY, MS_RDONLY, STRV_MAKE("/run", "/sys"));
+        if (r < 0)
+                log_warning_errno(r, "Read-only bind remount failed, ignoring: %m");
+
+        (void) mount_nofollow_verbose(LOG_WARNING,
+                                      "tmpfs", "/tmp", "tmpfs",
+                                      MS_NOSUID|MS_NODEV,
+                                      "mode=01777" TMPFS_LIMITS_RUN);
```

Comment:
doesn't matter, but overmounting /tmp sounds like something that sooner or later should probably end up as new flag for safe_fork(), i.e. FORK_PRIVATE_TMP or so. Sounds like something we could use at other places too